app/models/post.rb

```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_keys: 'author_id'
+  has_many :posts, foreign_keys: 'author_id'
+  has_many :comments
```

Comment:
- [ ] Please ensure that you carefully review and verify the associations in your model classes. I noticed an invalid association in line `3` where you have defined a `has_many :posts` association within the Post model class itself. Since you are already within the Post model class, this association is unnecessary and can be removed.